### PR TITLE
Product change - code archives on test and live

### LINKED ISSUE #1390: Deploy tag checked out for backups in test and live
Re: [Restoring an Environment from a Backup
](https://pantheon.io/docs/articles/sites/backups/restoring-an-environment-from-a-backup/#restoring-the-codebase) 

## Issue Description
Archives created on the Test and Live environments now reflect the state of the codebase for the most recent deploy on the environment. This means that users do not have to search for tags when restoring codebase. 

## Suggested Resolution
Update instructions on how to restore an environment's codebase from Live/Test now that archives reflect most recent deployment automatically. 
- [x] Download codebase archive from Live or Test environment 
- [x] Push 'detached HEAD' state to target environment (Dev)
- [x] Update Docs
- [x] Add change summary to #1318 

Estimated time for task: ~30minutes - 1hour

### PR DESCRIPTION
**Pull request by rachelwhitton**
Closes #1390

## Effect
PR includes the following changes:
- Document product change: Code archives generated on Test and Live environments now automatically checkout the most recent deployment tag to reflect the state of code. 

## Remaining Work
- [x] Document the purpose of a backup - to restore in emergencies. 
- [x] Document how to restore Dev to the state of live, using `git reset <commitid> --hard` and force push, and then cloning the DB back from Live.

### CODE REVIEW

**Comment by rachelwhitton** [comment]
@bmackinney @ari-gold can you review? PR updated so that code archives are explained but not recommended when restoring the code base - instead docs recommend using `git revert` or `git reset` so code restores are done within a single local repository (the same clone used to manage code via git)

File: source/docs/articles/sites/code/hot-fixes.md [hunk 1]
```diff
@@ -9,7 +9,7 @@ For Experts only. You should not need to attempt this if you use [Multidev](/doc
 
 <div class="alert alert-danger" role="alert">
 <h4>Warning</h4>
-We do not recommend hotfixing. Hotfixes should be the exception, not the norm. Pushing a hotfix via Git is the only way to push code directly to Live without having to go through Dev and Test. Hotfixing is not a best practice and any damage to the source code will be the responsibility of the user and should be avoided whenever possible.
+We do not recommend hotfixing. Hotfixes should be the exception, not the norm. Pushing a hotfix via Git is the only way to push code directly to Live without having to go through Dev and Test. Hotfixing is not a best practice and any damage to the source code will be the responsibility of the user and should be avoided whenever possible. We recommend restoring Dev's codebase as an alternative, see <a href="/docs/articles/sites/backups/restoring-an-environment-from-a-backup/#restore-code">Restoring an Environment from a Backup</a>.
```
Comment (ari-gold) [inline source/docs/articles/sites/code/hot-fixes.md]: Hotfix doc does not need an update.

**Comment by ari-gold** [comment]
+1 Thanks @rachelwhitton!

File: source/docs/articles/sites/backups/restoring-an-environment-from-a-backup.md [hunk 2]
```diff
@@ -23,65 +23,44 @@ When a restore starts, it is placed in a queue and executed. Depending on the si
...
+### Restore Code
+Code archives contain the full remote Git repository and reflect the state of code for the given environment. Backups created on the Test and Live environments automatically checkout the [`git tag`](https://git-scm.com/book/en/v2/Git-Basics-Tagging) associated with the most recent deployment. However, if you would like to rewind an environment's codebase to a previous state we recommend using `git revert` or `git reset` instead of a code archive.
 
-## Import Existing Content
+#### Revert Commits and Preserve History
+This method is recommended for distributed teams working collaboratively. To undo commits while preserving the site's Git history:
 
-Once you have the downloaded parts and/or links, navigate to the target environment and click the **Workflow** tab.  
+1. Identify the commit you want to undo using the commit history provided in the site Dashboard or by reviewing `git log` locally
+2. Copy the commit ID
```
Comment (bmackinney) [inline source/docs/articles/sites/backups/restoring-an-environment-from-a-backup.md]: Consider adding a small screenshot of a commit, highlighting the ID in the dashboard. 
image

**Pull request merged**